mvcc/kvstore.go

```diff
@@ -171,6 +171,8 @@ func (s *store) Hash() (hash uint32, revision int64, err error) {
 	h, err := s.b.Hash(DefaultIgnores)
 
 	hashSec.Observe(time.Since(start).Seconds())
+	s.revMu.RLock()
+	defer s.revMu.RUnlock()
```

Comment:
As discussed, adding lock here is not going to help much. For now, can we just add a TODO here as reminder?

Reply:
I read your discussion but I am not familiar with these packages, so it's quite possible that my understanding is wrong. 
Do you think the following is appropriate? 
```Go
// TODO: hash and revision could be inconsistent, due to absence of s.revMu.Lock() 
//  at the beginning of function, which is costly
func (s *store) Hash() (hash uint32, revision int64, err error) {
```

Reply:
Either adding `s.revMu.RLock()` or `s.mu.Lock()` can fix it. `s.revMu.RLock()` is likely better. There is a less costly way of fixing this, which is to get revision from the opened backend read transaction in s.b.Hash(). Probably just say: 

"TODO: hash and revision could be inconsistent, one possible fix is to add s.revMu.RLock() at the beginning of function, which is costly".